src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -219,6 +219,16 @@ void CodegenNeuronCppVisitor::print_function_or_procedure(const ast::Block& node
         printer->fmt_line("int ret_{} = 0;", name);
     }
 
+    if (node.is_procedure_block() || node.is_function_block()) {
```

Comment:
This is necessary because of the strange way NEURON treats global variables; in `hh.mod`, we have the following problem:
- the PROCEDURE `states` takes no args
- the PROCEDURE `rates` takes one arg, but which is actually _not_ an arg but the global voltage parameter `v`

On top of this, in the generated code, `v` is actually `v_unused`, hence the code here (`v_unused` is actually `naming::VOLTAGE_UNUSED_VARIABLE`).

Reply:
If I look at `nocmodl` generated code I get:
```
static int  rates ( _internalthreadargsprotocomma_ double _lv ) {
  // ..
  _lalpha = .1 * vtrap ( _threadargscomma_ - ( _lv + 40.0 ) , 10.0 ) ;
  //..
}

static void {_hoc,_npy}_rates(void) {
  // ..
  rates ( _threadargscomma_ *getarg(1) );
}

static void initmodel(_internalthreadargsproto_) {
  // ... 
  rates ( _threadargscomma_ v ) ;
}
```
which to me reads like the argument of `rates` is a regular variable, that happens to be called `v` (and shadows the node value `v`).

Similarly for CoreNEURON:
```
inline int rates_hodhux(int id, int pnodecount, hodhux_Instance* inst, double* data, const Datum* indexes, ThreadDatum* thread, NrnThread* nt, double v, double arg_v);
```
(which is then never used).

Reply:
> the argument of `rates` is a regular variable, that happens to be called `v` (and shadows the node value `v`).

That is correct, and the reason in nocmodl that the arg is translated as _lv using the pattern _larg.

Reply:
So what's the recommended fix here exactly?